src/test/java/com/example/weather/client/mappers/WeatherMapperTest.java

```diff
@@ -0,0 +1,43 @@
+package com.example.weather.client.mappers;
+
+import com.example.weather.client.models.dto.WeatherDto;
+import com.example.weather.client.models.entity.Weather;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.co.jemos.podam.api.PodamFactory;
+import uk.co.jemos.podam.api.PodamFactoryImpl;
+
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+class WeatherMapperTest {
+
+    private static PodamFactory FACTORY = new PodamFactoryImpl();
+    private static final WeatherDto WEATHER_DTO = FACTORY.manufacturePojo(WeatherDto.class);
+    private static final List<WeatherDto> WEATHER_DTO_LIST = List.of(WEATHER_DTO);
+    private WeatherMapper weatherMapper = new WeatherMapper();
+
+    @Test
+    void testToEntity() {
+        var weather = weatherMapper.toEntity(WEATHER_DTO);
+        assertThat(weather).isNotNull();
+        assertEquals(weather.getParameters(), WEATHER_DTO.getMain());
+        assertEquals(weather.getDescription(), WEATHER_DTO.getDescription());
+    }
+
+    @Test
+    void testToEntityList() {
+        var weathers = weatherMapper.toEntityList(WEATHER_DTO_LIST);
```

Comment:
(sorry for being grammar-nazi 😉) weather doesn't have plural form, so better name'd be `weatherList`